source/particle/integrator/rk_2.cc

```diff
@@ -0,0 +1,141 @@
+          {
+            const types::particle_index id_num = it->second.get_id();
+            const Point<dim> loc = it->second.get_location();
+            if (step == 0)
+              {
+                loc0[id_num] = loc;
+                it->second.set_location(loc + 0.5 * dt * (*old_vel));
+              }
+            else if (step == 1)
+              {
+                it->second.set_location(loc0[id_num] + dt * (*old_vel + *vel)/2.0);
+              }
+            else
+              {
+                Assert(false,
+                       ExcMessage("The RK2 integrator should never continue after two integration steps."));
+              }
+          }
+      }
+
+      template <int dim>
+      bool
+      RK2<dim>::new_integration_step()
+      {
+        if (step == 1) loc0.clear();
+        step = (step+1)%2;
+
+        // Continue until we're at the last step
+        return (step != 0);
+      }
+
+      template <int dim>
+      unsigned int
+      RK2<dim>::get_data_size() const
+      {
+        // TODO: If we finished integration we do not need to transfer integrator data. Return 0 in that case.
+        return dim * sizeof(double);
+      }
+
+      template <int dim>
+      void
+      RK2<dim>::read_data(const void *&data,
+                          const types::particle_index id_num)
+      {
+        // TODO: If we finished integration we do not need to transfer integrator data. Return early in that case.
+
+        const double *integrator_data = static_cast<const double *> (data);
+
+        // Read location data
+        for (unsigned int i=0; i<dim; ++i)
+          loc0[id_num](i) = *integrator_data++;
+
+        data = static_cast<const void *> (integrator_data);
+      }
+
+      template <int dim>
+      void
+      RK2<dim>::write_data(void *&data,
+                           const types::particle_index id_num) const
+      {
+        // TODO: If we finished integration we do not need to transfer integrator data. Return early in that case.
+
+        double *integrator_data = static_cast<double *> (data);
+
+        // Write location data
+        const typename std::map<types::particle_index, Point<dim> >::const_iterator it = loc0.find(id_num);
+        for (unsigned int i=0; i<dim; ++i,++integrator_data)
+          *integrator_data = it->second(i);
+
+        data = static_cast<void *> (integrator_data);
+      }
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Integrator
+    {
+      ASPECT_REGISTER_PARTICLE_INTEGRATOR(RK2,
+                                          "rk2",
+                                          "Runge Kutta second order integrator, where "
+                                          "y_{n+1} = y_n + dt*v(0.5*k_1), k_1 = dt*v(y_n). "
+                                          "This scheme requires storing the original location, "
+                                          "and the read/write_data functions reflect this.")
```

Comment:
The last two lines are really just a note about the actual implementation, but they don't mean very much to the person trying to decide which integrator to use. I'd just remove them (and probably similarly from the other integrators that may have a similar statement).